### ISSUE TITLE
Add 'Get Help' button to training modules

### ISSUE THREAD

**Issue by ragesoss** [WikiEducationFoundation/WikiEduDashboard#5486]
### Current Behavior:

For logged in users, in `wiki_education` mode, the 'Get Help' button appears in the Course navbar (whenever a user is on a course page). However, when a student launches a training module from their course page, the Course navbar is not part of the training interface and there is no 'Get Help' button.

### Desired Behavior:

For logged in users, a 'Get Help' button should also appear when going through training modules.

### Additional context:

Currently, the training navbar is rendered server-side (while the main navbar at the top is rendered in React on the `nav_root` element, and the training slides are rendered in React below the training navbar on the `react_root` element).

The best option will probably be to refactor the training pages so that the training navbar also renders in React.

The behavior of the 'Get Help' button might need to be a little different, because on Course pages, it shows different options depending on the staff roles assigned to the course. It should probably just default to sending alerts to the Wiki Expert for the user's course (which could either be inferred server-side based on the user and which courses they are part of, or a course ID could be supplied). One way to get the relevant course would be to extract the slug from the `data-return-to` property of the React root (which is included in the slide_view html).

**Comment by mapra99** [comment]
@ragesoss hey! I'd like to look into this

**Comment by justiceotuya** [comment]
@ragesoss how do we navigate to the course page. cant seem to see that. also I have a training page that is not populated. is this the intended behaviour?

<img width="1800" alt="image" src="https://github.com/WikiEducationFoundation/WikiEduDashboard/assets/17678464/95555726-5d2e-4b23-938d-9d4e467cef10">

**Comment by ragesoss** [comment]
@justiceotuya see https://github.com/WikiEducationFoundation/WikiEduDashboard/blob/master/docs/user_roles.md#loading-training-modules

To enable the Get Help button on a course page, you also need to be in `wiki_education: true` mode (via application.yml).

**Comment by sharon-odhiambo** [comment]
Hello, I'll be working on this issue.

**Comment by Ritesh17rb** [comment]
Has this issue been resolved @ragesoss ?

**Comment by ragesoss** [comment]
@Ritesh17rb no, this is still open.

**Comment by om-chauhan1** [comment]
Working on this

**Comment by JiyaGupta-cs** [comment]
@ragesoss I guess the ```data-return-to``` way is not correct for all the possibilities, since if the user doesnt get to that training via any course then it won't include the course,
In [this](https://github.com/WikiEducationFoundation/WikiEduDashboard/pull/5775#issuecomment-2181390149), maybe you entered here directly from the ```/training``` url that's why you didn't see the button to send alert like [this](https://github.com/WikiEducationFoundation/WikiEduDashboard/pull/5775#issuecomment-2061192565).
So, I guess we need to retrieve from the backend all the courses of which the user is part of and send all the course ids along with that message.
Am I on the right path ?

**Comment by JiyaGupta-cs** [comment]
> Hmm... that seems reasonable, to make a call to the backend to find the appropriate course whenever the Get Help button is clicked. Probably just using the most recent course for the user would be fine.

Yeah. Going with that, if there any function already available to get the recent course from the backend which I can reuse?

**Comment by JiyaGupta-cs** [comment]
> Not that I can think of.

Then what should be my approach now to solve this.
Because I dont think if we send all the courses from ```user.courses``` would be beneficial to create an alert
Or we can just send the alert without associating any course so that the query is sent to the WikimediaHelpUser

**Comment by Formasitchijoh** [comment]
Hi @ragesoss , should the` Get Help `button in training have the exact same functionality as the one in the course nav?

You mentioned "it should probably just default to sending alerts to the Wiki Expert for the user's course" ,  are those alerts going to the course instructor or a designated Wiki Expert? Do we have specific expert data we'd use?

One related concern: not every course has a user with WIKI_ED_STAFF_ROLE assigned. The original Get Help button on the course nav handles this by hiding the contact section when no staff are found , our training version would do the same. Should we instead fall back to a global help contact `(e.g. SpecialUsers.technical_help_staff)` so students always have someone to reach, or is silently hiding the contact section the right behaviour?

A second concern with the "most recent course" approach: if a student is enrolled in multiple courses, the most recent one may not be the course they're taking the training for. My suggestion would be ,  when Get Help is clicked from the top-level `/training` page, show the user a list of their active courses (most recent first) so they can pick the right one, rather than defaulting to something that could be wrong
<img width="1680" height="1050" alt="Image" src="https://github.com/user-attachments/assets/e641aae7-2138-4f3d-a4f2-5c984a843ff1" />

**Comment by ragesoss** [comment]
I think it will be very rare for a student to be doing training for a course other than their most recently one, so I think using the most recent one to set the relevant users is fine. For ClassroomProgramCourses in production, there should basically always be a Wiki Education staff role assigned, but I think if the button loads the user's most recent course and just renders the button with the same behavior as on the course page itself (in terms of which options to hide/show), this will work well.

**Comment by GonzaFP** [comment]
@Formasitchijoh are you working on this? I would like to give it a try!

**Comment by Formasitchijoh** [comment]
Hi @GonzaFP  you can give it a try.

**Comment by GonzaFP** [comment]
Hello @ragesoss @Formasitchijoh 
I am nearly done implementing the training nav. Before I do a PR, I would like to know how the training nav should behave on different screen sizes.
Thanks

<img width="1366" height="587" alt="Image" src="https://github.com/user-attachments/assets/6634e7dd-4d09-478b-b5b1-d5cc83f0d0d6" />

**Comment by ragesoss** [comment]
I think the course page nav is a reasonable model for how the training nav should behave. It wraps to two lines on mobile, with the Get Help button still visible (at the end of the second row).
